library/core/src/ops/control_flow.rs

```diff
@@ -81,6 +82,35 @@ impl<B, C> Try for ControlFlow<B, C> {
     }
 }
 
+#[unstable(feature = "try_trait_v2", issue = "84277")]
+impl<B, C> ops::TryV2 for ControlFlow<B, C> {
+    type Output = C;
+    type Residual = ControlFlow<B, convert::Infallible>;
```

Comment:
What's the reasoning for using `Infallible` over `!`?

Reply:
I don't want to accidentally introduce (another) way to refer to real-`!` if `never_type` hasn't stabilized yet but someone gets a hold of this residual type's name.  Like if someone finds a way to use a trick like
```
trait Foo { type Bar; }
impl<T, E> Foo for Result<T, E> { type Bar = T; }
type RealNever = <Result<(),()>::Residual as Foo>::Bar;
```
(I don't know how they'd mention `::Residual` like that, but with inference I worry they'd find a clever way.)

Hopefully the work on never type stabilization will mean soon that this just becomes `!` before that can become an issue, but I've added a checkbox in https://github.com/rust-lang/rust/issues/84277#issue-860487457 to make sure we look at this before stabilizing `try_trait_v2`.